src/tiff.imageio/tiffinput.cpp

```diff
@@ -1428,7 +1428,8 @@ TIFFInput::read_native_scanline(int subimage, int miplevel, int y, int z,
             ImageSpec dummyspec;
             int old_subimage = current_subimage();
             int old_miplevel = current_miplevel();
-            if (!close() || !open(m_filename, dummyspec)
+            close_tif();
```

Comment:
Given how tricky this was to track down -- at least leave a comment about why you call `close_tif` vs. `close()` here?

And surely this deserves a unit test? :smile:

Reply:
Yeah, will do...